### PR TITLE
CHI-1720: refactor ssm-cache to allow building cache of individual params

### PR DESCRIPTION
**Pull request by robert-bo-davis**
## Description

This change allows ssm-cache to build a cache at call time instead of building a full cache via getParametersByPath. It *should* work with either pattern now. See [the ssm-cache README](https://github.com/techmatters/hrm/blob/bo/feat/ssm_cache_refactor/packages/ssm-cache/README.md) for more info.

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
CHI-1720

### Verification steps

Run `npm run test`

### CODE REVIEW

**Comment by robert-bo-davis** [comment]
This one is tested and ready for review now as well.

File: packages/ssm-cache/index.ts [hunk 1]
```diff
@@ -46,31 +55,75 @@ export class SsmParameterNotFound extends Error {
...
+export const hasCacheExpired = () => !!(ssmCache.expiryDate && new Date() > ssmCache.expiryDate);
+
+export const isConfigNotEmpty = () => !!Object.keys(ssmCache.values).length;
+
+export const setCacheDurationMilliseconds = (cacheDurationMilliseconds: number) => {
+  ssmCache.cacheDurationMilliseconds = cacheDurationMilliseconds;
+};
+
+export const parameterExistsInCache = (name: string) =>
+  Object.prototype.hasOwnProperty.call(ssmCache.values, name) && ssmCache.values[name];
```
Comment (stephenhand) [inline packages/ssm-cache/index.ts:71]: Is the first part of the check necessary?

File: packages/ssm-cache/index.ts [hunk 2]
```diff
@@ -46,31 +55,75 @@ export class SsmParameterNotFound extends Error {
...
+  if (!parameterExistsInCache(name) || hasParameterExpired(ssmCache.values[name])) {
+    await loadParameter(name);
+  }
+
+  // If the cache still doesn't have the requested parameter, throw an error
+  if (!parameterExistsInCache(name)) {
+    throw new SsmParameterNotFound(`Parameter ${name} not found`);
+  }
+
+  return ssmCache?.values[name]?.value || '';
```
Comment (stephenhand) [inline packages/ssm-cache/index.ts:120]: Would ssmCache ever not be defined? We don't check for it in `parameterExistsInCache`

**Review by stephenhand — APPROVED** [review]
Nice rework - couple of nit Qs just to show I was paying attention really.

Not tested but code & tests LGTM

File: packages/ssm-cache/index.ts [hunk 3]
```diff
@@ -46,31 +55,74 @@ export class SsmParameterNotFound extends Error {
...
+  if (!parameterExistsInCache(name) || hasParameterExpired(ssmCache.values[name])) {
+    await loadParameter(name);
+  }
+
+  // If the cache still doesn't have the requested parameter, throw an error
+  if (!parameterExistsInCache(name)) {
+    throw new SsmParameterNotFound(`Parameter ${name} not found`);
+  }
+
+  return ssmCache.values[name]?.value || '';
```
Comment (robert-bo-davis) [inline packages/ssm-cache/index.ts:119]: This line irritates me. parameterExists in Cache now catches falsy values but TS still makes me add the `|| ''` to the return. Is there some way around that inability to follow logic into validation that throw errors in TS? Or am missing some other caveat?

**Comment by robert-bo-davis** [comment]
Refactored a bit to fix both of the nits.

I also made [this change](https://github.com/techmatters/hrm/pull/308/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R83-R84) to the `test` script because I keep getting failing tests when switching between branches because dependant packages aren't automatically built by jest-ts.

One last question is cacheDuration. Should we increase it now that new ssm params that are added will be picked up? None of the SSM params currently being fetched will change very often and possibly never.

File: packages/ssm-cache/index.ts (hunk 1 above)
Comment (robert-bo-davis) [reply packages/ssm-cache/index.ts]: Good catch. This logic was flawed in several ways. Refactored to make it treat falsy values as missing as well.

**Review by stephenhand — APPROVED** [review]
LGTM

**Pull request merged**